relay-event-normalization/src/event.rs

```diff
@@ -166,151 +162,141 @@ impl<'a> Default for NormalizeProcessorConfig<'a> {
 
 /// Normalizes an event, rejecting it if necessary.
 ///
-/// The normalization consists of applying a series of transformations on the
-/// event payload based on the given configuration.
+/// Normalization consists of applying a series of transformations on the event
+/// payload based on the given configuration.
 ///
 /// The returned [`ProcessingResult`] indicates whether the passed event should
 /// be ingested or dropped.
-#[derive(Debug, Default)]
-pub struct NormalizeProcessor<'a> {
-    /// Configuration for the normalization steps.
-    config: NormalizeProcessorConfig<'a>,
-}
-
-impl<'a> NormalizeProcessor<'a> {
-    /// Returns a new [`NormalizeProcessor`] with the given config.
-    pub fn new(config: NormalizeProcessorConfig<'a>) -> Self {
-        Self { config }
-    }
-}
-
-impl<'a> Processor for NormalizeProcessor<'a> {
-    fn process_event(
-        &mut self,
-        event: &mut Event,
-        meta: &mut Meta,
-        _: &ProcessingState<'_>,
-    ) -> ProcessingResult {
-        if self.config.is_renormalize {
-            return Ok(());
-        }
-
-        // Validate and normalize transaction
-        // (internally noops for non-transaction events).
-        // TODO: Parts of this processor should probably be a filter so we
-        // can revert some changes to ProcessingAction)
-        let mut transactions_processor = transactions::TransactionsProcessor::new(
-            self.config.transaction_name_config.clone(),
-            self.config.transaction_range.clone(),
-        );
-        transactions_processor.process_event(event, meta, ProcessingState::root())?;
-
-        // Check for required and non-empty values
-        schema::SchemaProcessor.process_event(event, meta, ProcessingState::root())?;
-
-        TimestampProcessor.process_event(event, meta, ProcessingState::root())?;
+pub fn normalize_event(
+    event: &mut Annotated<Event>,
+    config: &NormalizationConfig,
+) -> ProcessingResult {
+    let Annotated(Some(ref mut event), ref mut meta) = event else {
+        return Ok(());
+    };
 
-        // Process security reports first to ensure all props.
-        normalize_security_report(event, self.config.client_ip, &self.config.user_agent);
+    let is_renormalize = config.is_renormalize;
```

Comment:
seems a bit redundant imo

Reply:
I find this a bit easier to read and reuse, but I'll revisit this decision in the future if it's no longer applicable.